### DISCUSSION TITLE
Best practice: How to structure a complex app?

### DISCUSSION THREAD

**Discussion by Robert-M-Muench** [wailsapp/wails#909] [General]
After playing around with Wails V2 for a while now, I'm wondering what's best practice for complex apps?

* Should UI interactivity be done with something like react?
* Should any interaction be routed to the Go side ASAP and the HTML, JS, CSS part be kept to a minimum?

If we see Wails as "add a UI layer to your (non-UI) Go application" I think everything UI related should go into the frontend. The Go part will be the backend, the headless version if your application.

If we want to minimize the zoo of technologies used, we should do most stuff on the Go side and use the frontend as a generic UI layer that can deal with declarative input (HTML, JS, CSS).

I would be interested in your feedback and from people having used Wails to create more complex apps.

**Comment by leaanthony** [comment]
Well that just depends. For me, the frontend just reflects the state in Go. For others, they may see Go as a shell/persistence layer. There's no hard and fast rules.

**Reply by leaanthony** [reply]
An exception to this might be the case of a text editor where all that text may live just in the frontend. This is an edge case.
